src/main/java/org/trigon/hardware/misc/leds/AddressableLEDStrip.java

```diff
@@ -0,0 +1,216 @@
+package org.trigon.hardware.misc.leds;
+
+import com.ctre.phoenix.led.LarsonAnimation;
+import edu.wpi.first.wpilibj.AddressableLED;
+import edu.wpi.first.wpilibj.AddressableLEDBuffer;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.util.Color;
+import org.trigon.hardware.RobotHardwareStats;
+
+import java.util.function.Supplier;
+
+/**
+ * A LED strip that is controlled by an AddressableLED.
+ */
+public class AddressableLEDStrip extends LEDStrip {
+    private static AddressableLED LED;
+    private static AddressableLEDBuffer LED_BUFFER;
+
+    private final boolean shouldOnlyRunInSimulation;
+    private int lastBreatheLED;
+    private double lastLEDAnimationChangeTime = 0;
+    private double rainbowFirstPixelHue = 0;
+    private boolean isLEDAnimationChanged = false;
+    private int amountOfColorFlowLEDs = 0;
+
+    /**
+     * Sets the AddressableLED instance to be used for controlling the LED strip. Must be set before using any LED strips.
+     * The LED instance should be configured before being set, however it does not need to be started.
+     *
+     * @param led the LED instance to be used
+     */
+    public static void setAddressableLED(AddressableLED led) {
+        if (LED == null) {
+            LED = led;
+            LED.start();
+        }
+    }
+
+    /**
+     * Sets the AddressableLEDBuffer instance to be used for controlling the LED strip. Must be set before using any LED strips.
+     * The LED buffer instance must be configured before being set.
+     *
+     * @param ledBuffer the LED buffer instance to be used
+     */
+    public static void setAddressableLEDBuffer(AddressableLEDBuffer ledBuffer) {
+        if (LED_BUFFER == null)
+            LED_BUFFER = ledBuffer;
+    }
+
+    /**
+     * Constructs a new AddressableLEDStrip. Before any commands are sent to the LED strip, the setAddressableLED and setAddressableLEDBuffer methods must be called.
+     *
+     * @param inverted     whether the LED strip is inverted
+     * @param numberOfLEDs the amount of LEDs in the strip
+     * @param indexOffset  the offset of the first LED in the strip
+     */
+    public AddressableLEDStrip(boolean inverted, int numberOfLEDs, int indexOffset) {
+        this(inverted, numberOfLEDs, indexOffset, false);
+    }
+
+    AddressableLEDStrip(boolean inverted, int numberOfLEDs, int indexOffset, boolean shouldOnlyRunInSimulation) {
+        super(inverted, numberOfLEDs, indexOffset);
+        this.shouldOnlyRunInSimulation = shouldOnlyRunInSimulation;
+        resetLEDSettings();
+    }
+
+    @Override
+    public void periodic() {
+        if (shouldOnlyRunInSimulation && !RobotHardwareStats.isSimulation())
```

Comment:
why this variable 
just use the normal isSimulation without the other variable

Reply:
Because if we're using CANdle then we only want the AddressableLED periodic to run in sim, but usually we want it to run in both